Add `proxysql_mysql_hostgroup_attributes`.

burnison opened the pull request:
##### SUMMARY

This PR adds support for managing [`mysql_hostgroup_attributes`].

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME

`proxysql_mysql_hostgroup_attributes`

##### ADDITIONAL INFORMATION

The [`mysql_hostgroup_attributes`] table allows for per-hostgroup overrides, which change the behaviour of a specific hostgroup.

For example, setting the `max_num_online_servers` value for a host group will change ProxySQL's behaviour when the number of `ONLINE` servers exceed this value:

```
- name: "test_create | create"
  proxysql_mysql_hostgroup_attributes:
    login_user: admin
    login_password: admin
    save_to_disk: true
    load_to_runtime: false
    hostgroup_id: 1
    max_num_online_servers: 1
  register: create
```

Resulting in the following change set:
```
changed: [testhost] => {
    "changed": true,
    "invocation": {
        "module_args": {
            "autocommit": -1,
            "comment": "",
            "config_file": "",
            "connection_warming": 0,
            "free_connections_pct": 10,
            "hostgroup_id": 1,
            "hostgroup_settings": "",
            "ignore_session_variables": "",
            "init_connect": "",
            "load_to_runtime": false,
            "login_host": "127.0.0.1",
            "login_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "login_port": 6032,
            "login_unix_socket": null,
            "login_user": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "max_num_online_servers": 1,
            "multiplex": 1,
            "save_to_disk": true,
            "servers_defaults": "",
            "state": "present",
            "throttle_connections_per_sec": 1000000
        }
    },
    "msg": "Added entry to mysql_hostgroup_attributes",
    "mysql_hostgroup_attributes": {
        "autocommit": "-1",
        "comment": "",
        "connection_warming": "0",
        "free_connections_pct": "10",
        "hostgroup_id": "1",
        "hostgroup_settings": "",
        "ignore_session_variables": "",
        "init_connect": "",
        "max_num_online_servers": "1",
        "multiplex": "1",
        "servers_defaults": "",
        "throttle_connections_per_sec": "1000000"
    },
    "state": "present"
}
```


[`mysql_hostgroup_attributes`]: https://proxysql.com/documentation/main-runtime/#mysql_hostgroup_attributes

burnison committed:
Add `proxysql_mysql_hostgroup_attributes`.

The `[mysql_hostgroup_attributes]` table allows for per-hostgroup
overrides, which change the behaviour of a specific hostgroup.

For example, setting the `max_num_online_servers` value for a host group
will change ProxySQL's behaviour when the number of `ONLINE` servers
exceed this value. Similarly, multiplexing may be enabled or disabled on
a per-hostgroup basis.

This PR adds support for managing `mysql_hostgroup_attributes`.

[mysql_hostgroup_attributes]: https://proxysql.com/documentation/main-runtime/#mysql_hostgroup_attributes

Andersson007 commented on plugins/modules/proxysql_mysql_hostgroup_attributes.py:
```diff
@@ -0,0 +1,437 @@
+     hostgroup-specific configurations, like `max_num_online_servers`, which
+     change the behaviour of the given hostgroup in ways that are not otherwise
+     possible.
+
```
```suggestion
version_added: '1.7.0'

```

Andersson007 commented on changelogs/fragments/180_add_mysql_hostgroup_attributes.yml:
```diff
@@ -0,0 +1,2 @@
+minor_changes:
```
```suggestion
minor_changes:
```
please remove the fragment. our changelog generator tool will generate an entry automatically based on module's description and version_added field

Andersson007 commented on plugins/modules/proxysql_mysql_hostgroup_attributes.py:
```diff
@@ -0,0 +1,437 @@
+- community.proxysql.proxysql.managing_config
+- community.proxysql.proxysql.connectivity
+notes:
+- Supports C(check_mode).
```
```suggestion
notes:
- Supports C(check_mode).
```
Use the `attributes:` field instead. Grep through, say, community.postgresql collection or see on docs.ansible.com module format and documentation page

burnison replied:
Thank for the tip here. It looks like the rest of this collection uses `notes` for the same purpose, but I've looked into some other modules to find the change.  I'll change this one in this PR and try to follow-up with another PR for the other plugins.

Andersson007 commented on plugins/modules/proxysql_mysql_hostgroup_attributes.py:
```diff
@@ -0,0 +1,437 @@
+'''
+
+EXAMPLES = '''
+---
```
```suggestion
```

Andersson007 commented on plugins/modules/proxysql_mysql_hostgroup_attributes.py:
```diff
@@ -0,0 +1,437 @@
+# This example removes a hostgroup override using the credentials supplied in
+# a configuration file.
+
+- name: Remove hostgroup overrides
```
```suggestion
- name: Remove hostgroup overrides
```
I see a lot of options in DOCUMENTATION. Should we add a few more examples?

burnison replied:
I've added several more usage examples.

Andersson007 reviewed (commented):
@burnison thanks for contributing the module!

burnison committed:
More documentation for `hostgroup_attributes`.

This commit adds additional documentation for the
`proxysql_mysql_hostgroup_attributes` plugin.

Co-authored-by: Andrew Klychkov <aaklychkov@mail.ru>

Andersson007 commented:
@burnison thanks for the changes!
I've asked @markuman to take a look as I'm not a user. Let's wait for his feedback for a few days.
If no objections/feedback and if I forget, please ping me here at the end of the week to merge the PR
Thank you

markuman commented:
LGTM

Andersson007 commented:
@burnison thanks for contributing the module!
@markuman thanks for the contribution!
